pkg/operator/controller/ingress/poddisruptionbudget_test.go

```diff
@@ -87,3 +87,66 @@ func TestDesiredPodDisruptionBudget(t *testing.T) {
 		}
 	}
 }
+
+func TestPodDisruptionBudgetChange(t *testing.T) {
+	two := int32(2)
+	three := int32(3)
+	trueVar := true
+
+	testCases := []struct {
+		description string
+		mutate      func(ic *operatorv1.IngressController)
+		expect      bool
+	}{
+		{
+			description: "if nothing changes",
+			mutate:      func(_ *operatorv1.IngressController) {},
+			expect:      false,
+		},
+		{
+			description: "if replicas changes from 2 to 3",
+			mutate: func(ic *operatorv1.IngressController) {
+				ic.Spec.Replicas = &three
+			},
+			expect: true,
+		},
+	}
+
+	// Set up the original ingress controller with 2 replicas
+	ic := &operatorv1.IngressController{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+		},
+		Spec: operatorv1.IngressControllerSpec{
+			Replicas: &two,
+		},
+	}
+	deploymentRef := metav1.OwnerReference{
+		APIVersion: "apps/v1",
+		Kind:       "Deployment",
+		Name:       "router-default",
+		UID:        "1",
+		Controller: &trueVar,
+	}
+	// Get the original pdb based on the ingress controller and deployment
+	_, originalPdb, err := desiredRouterPodDisruptionBudget(ic, deploymentRef)
+	if err != nil {
+		t.Errorf("expected setup to succeed, but there was a failure: %v", err)
+	}
+
+	for _, tc := range testCases {
+		// Change the ingress controller and check the resulting pdb
+		tc.mutate(ic)
+		_, mutatedPdb, err := desiredRouterPodDisruptionBudget(ic, deploymentRef)
```

Comment:
Using `desiredRouterPodDisruptionBudget` to generate the test input seems a bit odd to me.  This is a unit test for verifying that `podDisruptionBudgetChanged` behaves correctly, and we already have another unit test for `desiredRouterPodDisruptionBudget`.  Any reason not to open-code the PDB?  Something like the following (which also uses `t.Run`):
```go
func TestPodDisruptionBudgetChange(t *testing.T) {
	pointerTo := func(ios intstr.IntOrString) *intstr.IntOrString { return &ios }
	testCases := []struct {
		description string
		mutate      func(pdb *policyv1.PodDisruptionBudget)
		expect      bool
	}{
		{
			description: "if nothing changes",
			mutate:      func(_ *policyv1.PodDisruptionBudget) {},
			expect:      false,
		},
		{
			description: "if MaxUnavailable changes from 25% to 50%",
			mutate: func(pdb *policyv1.PodDisruptionBudget) {
				pdb.Spec.MaxUnavailable = pointerTo(intstr.FromString("50%"))
			},
			expect: true,
		},
	}

	pdb := policyv1.PodDisruptionBudget{
		ObjectMeta: metav1.ObjectMeta{
			Name:      "router-defualt",
			Namespace: "openshift-ingress",
		},
		Spec: policyv1.PodDisruptionBudgetSpec{
			MaxUnavailable: pointerTo(intstr.FromString("25%")),
		},
	}

	for _, tc := range testCases {
		t.Run(tc.description, func(t *testing.T) {
			originalPDB := pdb.DeepCopy()
			mutatedPDB := pdb.DeepCopy()
			tc.mutate(mutatedPDB)
			if changed, updatedPDB := podDisruptionBudgetChanged(originalPDB, mutatedPDB); changed != tc.expect {
				t.Errorf("expected %t, got %t", tc.expect, changed)
			} else if changed {
				if changedAgain, _ := podDisruptionBudgetChanged(mutatedPDB, updatedPDB); changedAgain {
					t.Error("podDisruptionBudgetChanged does not behave as a fixed point function")
				}
			}
		})
	}
}
```

Reply:
No reason, but is there a technical issue with my proposed fix?

Reply:
It just seems a little confusing and more complicated than necessary.  It works, though.